bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/service/StateDescriptionServiceImplTest.java

```diff
@@ -0,0 +1,265 @@
+
+		item = new NumberItem(ITEM_NAME);
+		item.setStateDescriptionService(mergingService);
+		StateDescription finalStateDescription = item.getStateDescription();
+
+		assertThat(finalStateDescription.getMinimum(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_MIN_VALUE));
+		assertThat(finalStateDescription.getMaximum(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_MAX_VALUE));
+		assertThat(finalStateDescription.getStep(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_STEP));
+		assertThat(finalStateDescription.getPattern(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_PATTERN));
+		assertThat(finalStateDescription.isReadOnly(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_IS_READONLY));
+		assertThat(finalStateDescription.getOptions(), is(STATE_DESCRIPTION_PROVIDER_DEFAULT_OPTIONS));
+	}
+
+	@Test
+	public void testMinValueMaxValueStepAndPatternTwoDescriptionProviders() {
+		int stateDescriptionProvider1ServiceRanking = -1;
+		int stateDescriptionProvider2ServiceRanking = -2;
+
+		BigDecimal stateDescriptionProvider1MinValue = new BigDecimal("-1");
+		BigDecimal stateDescriptionProvider2MinValue = new BigDecimal("-2");
+
+		BigDecimal stateDescriptionProvider1MaxValue = new BigDecimal("-1");
+		BigDecimal stateDescriptionProvider2MaxValue = new BigDecimal("-2");
+
+		BigDecimal stateDescriptionProvider1Step = new BigDecimal("-1");
+		BigDecimal stateDescriptionProvider2Step = new BigDecimal("-2");
+
+		String stateDescriptionProvider1Pattern = "pattern1";
+		String stateDescriptionProvider2Pattern = "pattern2";
+
+		StateDescriptionProvider stateDescriptionProvider1 = mock(StateDescriptionProvider.class);
+		StateDescriptionProvider stateDescriptionProvider2 = mock(StateDescriptionProvider.class);
+
+		when(stateDescriptionProvider1.getRank()).thenReturn(stateDescriptionProvider1ServiceRanking);
+		when(stateDescriptionProvider2.getRank()).thenReturn(stateDescriptionProvider2ServiceRanking);
+
+		StateDescription stateDescription1 = new StateDescription(stateDescriptionProvider1MinValue,
+				stateDescriptionProvider1MaxValue, stateDescriptionProvider1Step, stateDescriptionProvider1Pattern,
+				false, null);
+		StateDescription stateDescription2 = new StateDescription(stateDescriptionProvider2MinValue,
+				stateDescriptionProvider2MaxValue, stateDescriptionProvider2Step, stateDescriptionProvider2Pattern,
+				false, null);
+
+		when(stateDescriptionProvider1.getStateDescription(ITEM_NAME, null)).thenReturn(stateDescription1);
+		when(stateDescriptionProvider2.getStateDescription(ITEM_NAME, null)).thenReturn(stateDescription2);
+
+		mergingService.addStateDescriptionProvider(stateDescriptionProvider1);
+		mergingService.addStateDescriptionProvider(stateDescriptionProvider2);
+		item = new NumberItem(ITEM_NAME);
+		item.setStateDescriptionService(mergingService);
+
+		StateDescription finalStateDescription = item.getStateDescription();
+
+		assertThat(
+				"expected minimum to be equal to the minumum from provider one because it has higher service ranking",
```

Comment:
Can you please remove the explanation string in the `assertThat` calls because they are hard to keep in sync and most of the time they are either stating the obvious or if they are meant as a error message for the failing test they are written to state the opposite of what the test actually checks.